core/src/data/dataSource.h

```diff
@@ -68,13 +68,24 @@ class DataSource {
         return url;
     }
 
-    std::map< TileID, std::shared_ptr<TileData> > m_tileStore; // Map of tileIDs to data for that tile
-    
-    std::string m_name; // Name used to identify this source in the style sheet
+    // Name used to identify this source in the style sheet
+    std::string m_name;
+
+    // URL template for requesting tiles from a network or filesystem
+    std::string m_urlTemplate;
+
+    // Used to ensure safe access from async loading threads
+    std::mutex m_mutex;
 
-    std::mutex m_mutex; // Used to ensure safe access from async loading threads
+    // LRU in-memory cache for raw tile data
+    using CacheEntry = std::pair<TileID, std::shared_ptr<std::vector<char>>>;
+    using CacheList = std::list<CacheEntry>;
+    using CacheMap = std::unordered_map<TileID, typename CacheList::iterator>;
```

Comment:
Instead of using 2 data containers, we can do away with using a single std::vector of pairs.

Why I suggest this:
- single container to hold our data, instead of 2 containers (and potential issues with using iterator as a MapValue).
- faster iteration on vectors over list
- for accessing we can do linear search first or binary search depending on how many tiles we have to search for in the 32MB cached memory.